### PR TITLE
fix(react-text): Update typography wrappers to latest naming

### LINKED ISSUE #22700: Misnaming of `react-text` wrapper components
### Describe the issue:

Due to the renaming of Fluent typography presets, `react-text` wrapper component names no longer match with the design.

#### Existing components
Headline -> Subtitle1
SubHeadline -> Subtitle2
Body -> Body1
Caption -> Caption1

#### Missing components
Caption2

### PR DESCRIPTION
**Pull request by andrefcdias**
## Current Behavior

We currently have these typography wrappers that no longer match the naming used by design:
Headline, SubHeadline, Body, Caption

## New Behavior

Components have been renamed to match the new design specs:

Headline -> Subtitle1
SubHeadline -> Subtitle2
Body -> Body1
Caption -> Caption1

New component `Caption2` was also added.

Also removes the deprecated `*ClassName` to help with the merging of #23026.

## Related Issue(s)

Fixes #22700

---

Apologies for the large PR, have a cute dog to keep you motivated through your journey:

extremely cute dog lying on a bed

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/36a89f225774d37710b772eb0e0a467c0e53217e'>36a89f225774d37710b772eb0e0a467c0e53217e</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/Avatar-mount_base.html">1096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/Avatar-mount.html">1067</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/Button-mount_base.html">681</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/Button-mount.html">672</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProvider-mount_base.html">2226</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProvider-mount.html">2229</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-mount_base.html">362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-mount.html">350</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">288</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">296</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">382</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">373</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/MakeStyles-mount_base.html">1891</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22934/perf-test-react-components/MakeStyles-mount.html">1861</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-text/src/components/Caption2/Caption2.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { makeStyles } from '@griffel/react';
+import { createWrapper, TextWrapperProps } from '../wrapper';
+import { SlotClassNames } from '@fluentui/react-utilities';
+import { TextSlots } from '../Text/Text.types';
+import { typographyStyles } from '@fluentui/react-theme';
+
+/**
+ * @deprecated Use `caption2ClassNames.root` instead.
+ */
+export const caption2ClassName = 'fui-Caption2';
```
Comment (andrefcdias) [inline packages/react-components/react-text/src/components/Caption2/Caption2.tsx:11]: Should we ship the deprecated `className` for new components too? Conformance is currently enforcing it; I assume for consistency reasons.

Comment (Hotell) [reply packages/react-components/react-text/src/components/Caption2/Caption2.tsx:11]: if we are introducing breaking change like this IMO this is good opportunity to get rid of it. what's the issue with conformance ?

**Review by Hotell — APPROVED** [review]

File: packages/react-components/react-text/src/components/Caption2/Caption2.tsx (hunk 1 above)
Comment (andrefcdias) [reply packages/react-components/react-text/src/components/Caption2/Caption2.tsx:11]: I must've been sleepwalking, there is no error anymore when removing it 😶 Pushing the change now

File: packages/react-components/react-text/src/components/Caption1/Caption1.tsx [hunk 2]
```diff
@@ -21,10 +21,10 @@ const useStyles = makeStyles({
 });
 
 /**
- * Text wrapper component for the Caption typography variant
+ * Text wrapper component for the Caption1 typography variant
  */
-export const Caption: FunctionComponent<TextWrapperProps> = createWrapper({
+export const Caption1: FunctionComponent<TextWrapperProps> = createWrapper({
```
Comment (miroslavstastny) [inline packages/react-components/react-text/src/components/Caption1/Caption1.tsx:26]: Besides the regular `Caption1`, typography spec also contains `Strong` and `Stronger` variants.
image
The same applies to some other Types as well. Do you plan to add those as well either through a separate components or a prop?

Comment (miroslavstastny) [reply packages/react-components/react-text/src/components/Caption1/Caption1.tsx:26]: Just check Text design spec as well, these Type presets are there as well.

File: packages/react-components/react-text/src/components/Caption2/Caption2.tsx (hunk 1 above)
Comment (andrefcdias) [reply packages/react-components/react-text/src/components/Caption2/Caption2.tsx]: image

https://uifabric.visualstudio.com/fabricpublic/_build/results?buildId=248744&view=logs&j=258ec178-2d8b-5611-7b9b-60c5c95dae55&t=8250ef03-1890-59ab-a6a8-5d0b4c2baa1b&l=1592

File: packages/react-components/react-text/src/components/Caption1/Caption1.tsx (hunk 2 above)
Comment (miroslavstastny) [reply packages/react-components/react-text/src/components/Caption1/Caption1.tsx:26]: huh, I see, thanks!

File: packages/react-components/react-text/src/components/Caption2/Caption2.tsx (hunk 1 above)
Comment (spmonahan) [reply packages/react-components/react-text/src/components/Caption2/Caption2.tsx]: You should remove `className`. We have another [issue to remove all the deprecated instances](https://github.com/microsoft/fluentui/issues/22132).

For conformance, you can disable the tests. Here's an example of how to do that: https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-spinbutton/src/components/SpinButton/SpinButton.test.tsx#L17

File: packages/react-components/react-text/src/stories/TextTypography.stories.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { Body, Caption, Display, Headline, LargeTitle, Title1, Title2, Title3, Subheadline } from '../index';
+import { Body1, Caption1, Display, Subtitle1, LargeTitle, Title1, Title2, Title3, Subtitle2, Caption2 } from '../index';
```
Comment (khmakoto) [inline packages/react-components/react-text/src/stories/TextTypography.stories.tsx:2]: _nit:_ Keep the alphabetical order

```suggestion
import { Body1, Caption1, Caption2, Display, LargeTitle, Title1, Title2, Title3, Subtitle1, Subtitle2 } from '../index';
```

File: packages/react-components/react-components/src/index.ts [hunk 4]
```diff
@@ -546,34 +546,36 @@ export type {
...
-  Caption,
+  Body1,
+  Caption1,
+  Caption2,
   Display,
-  Headline,
+  Subtitle1,
   LargeTitle,
-  Subheadline,
+  Subtitle2,
```
Comment (khmakoto) [inline packages/react-components/react-components/src/index.ts:555]: _nit:_ Keep these ordered alphabetically

```ts
LargeTitle,
Subtitle1,
Subtitle2,
```

File: packages/react-components/react-text/bundle-size/Text.Wrappers.fixture.js [hunk 5]
```diff
@@ -4,13 +4,13 @@ import {
...
-  Body,
-  Caption,
+  Subtitle1,
+  Subtitle2,
+  Body1,
+  Caption1,
 } from '@fluentui/react-text';
 
-console.log(Display, LargeTitle, Title1, Title2, Title3, Headline, Subheadline, Body, Caption);
+console.log(Display, LargeTitle, Title1, Title2, Title3, Subtitle1, Subtitle2, Body1, Caption1);
```
Comment (khmakoto) [inline packages/react-components/react-text/bundle-size/Text.Wrappers.fixture.js:13]:
```suggestion
console.log(Body1, Caption1, Display, LargeTitle, Subtitle1, Subtitle2, Title1, Title2, Title3);
```

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-text/src/components/Caption2/Caption2.tsx (hunk 1 above)
Comment (Hotell) [reply packages/react-components/react-text/src/components/Caption2/Caption2.tsx]: thanks @spmonahan !

**Comment by andrefcdias** [comment]
I've removed the deprecated `ClassName` from `react-text` as you asked for to help with the merging of https://github.com/microsoft/fluentui/pull/23026. Please review the changes before we merge this @spmonahan

**Review by spmonahan — CHANGES_REQUESTED** [review]
Just putting a block here so we can coordinate merging breaking changes

**Review by spmonahan — APPROVED** [review]

**Pull request merged**